### ISSUE TITLE
Bug: `no-invalid-regexp` not reporting repeted flag allowed by `allowConstructorFlags`

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#18748]
### Environment

Node version: latest
npm version: latest
Local ESLint version: latest
Global ESLint version: no
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint no-invalid-regexp: ["error", { "allowConstructorFlags": ["a"] }]*/

new RegExp('.', 'aa');
```

### What did you expect to happen?

report `aa` as invalid flag.

### What actually happened?

rule doesn't report this as invalid supplied flag as it does with other predefined flags.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW52YWxpZC1yZWdleHA6IFtcImVycm9yXCIsIHsgXCJhbGxvd0NvbnN0cnVjdG9yRmxhZ3NcIjogW1wiYVwiXSB9XSovXG5cbm5ldyBSZWdFeHAoJy4nLCAnYWEnKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [X] I am willing to submit a pull request for this issue.
